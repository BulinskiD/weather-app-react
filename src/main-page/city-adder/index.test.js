import React from 'react';
import ReactDOM from 'react-dom';
import CityAdder from './index';
import {shallow, mount} from "enzyme";
import LoadingContext from '../../context/loading-context';
import { act } from 'react-dom/test-utils';

describe('City adder', ()=>{
    const clickFn = jest.fn();
    const e = {preventDefault: ()=>{}};
    const unit = "imperial";

    beforeEach(()=>{
        clickFn.mockClear();
    });


    it('should render correctly with given props', ()=>{
        const component = shallow(<CityAdder unit={unit} onAddCity={clickFn} />);
        expect(component).toMatchSnapshot();
    });

    it('should render without crash without props', ()=>{
        const div = document.createElement('div');
        ReactDOM.render(<CityAdder />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('should call onAddCity func when form submitted with city name length >= 3', () => {
        act(() => {
            const component = shallow(<CityAdder unit={unit} onAddCity={clickFn} />);
            component.find('FormControl').props().onChange({target:{value:"sss"}});
            component.find('form').simulate('submit', e);
        });
        expect(clickFn).toBeCalledTimes(1);
    });

    it('should not call onAddCity func when form submitted with city name length < 3', () => {
        act(() => {
            const component = shallow(<CityAdder unit={unit} onAddCity={clickFn} />);
            component.find('FormControl').props().onChange({target:{value:"ss"}});
            component.find('form').simulate('submit', e);
        });
        expect(clickFn).toBeCalledTimes(0);
    });
});
