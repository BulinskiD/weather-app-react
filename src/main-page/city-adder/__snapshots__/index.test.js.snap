// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`City adder should render correctly with given props 1`] = `
<form
  onSubmit={[Function]}
>
  <Bootstrap(Row)
    className="mx-4"
  >
    <Bootstrap(InputGroup)>
      <FormControl
        aria-label="city"
        as="input"
        className="col-lg-8 col-12"
        onChange={[Function]}
        placeholder="Nazwa miasta"
        value=""
      />
      <Button
        active={false}
        className="offset-lg-1 col-lg-3 offset-0 col-12 my-3 my-lg-0"
        disabled={false}
        type="submit"
        variant="primary"
      >
        <ContextConsumer>
          <Component />
        </ContextConsumer>
        Dodaj
      </Button>
    </Bootstrap(InputGroup)>
  </Bootstrap(Row)>
</form>
`;
